syntax = "proto3";

message Command { //command that controls the status of esc current data transmittion to the main board will be received from the main board

    uint32 status = 1;
    
    enum ESCCurrentADCCommand{
        STOP = 0;
        START_CONTINIOUS = 1;
        START_ONETIME = 2;
      };

}

message ADC_data { //esc currents and batt voltage/current info will be transmitted to the main board

    float ch0_current = 1;
    float ch1_current = 2;
    float ch2_current = 3;
    float ch3_current = 4;
    float ch4_current = 5;
    float ch5_current = 6;
    float ch6_current = 7;
    float ch7_current = 8;

    float batt_voltage = 9;
    float batt_current = 10;
}

message PWM_ChannelData { //PWM periods need to drive escs will be received from the main board

    uint32 ch0 = 1;
    uint32 ch1 = 2;
    uint32 ch2 = 3;
    uint32 ch3 = 4;
    uint32 ch4 = 5;
    uint32 ch5 = 6;
    uint32 ch6 = 7;
    uint32 ch7 = 8;
}
